runtime/doc/news.txt

```diff
@@ -81,6 +81,8 @@ The following deprecated functions or APIs were removed.
 • Vimball support is removed.
   - :Vimuntar command removed.
 
+• Neovim QT is no longer shipped.
```

Comment:
I don't think this needs a news entry; this is more a project-level change that doesn't affect users _already having Neovim_.

Marking this as breaking is enough to put it front and center in the release notes.